api/v1alpha1/authorization_types.go

```diff
@@ -72,6 +96,36 @@ type Principal struct {
 	// JWT authentication in the same `SecurityPolicy`.
 	// +optional
 	JWT *JWTPrincipal `json:"jwt,omitempty"`
+
+	// Headers authorize the request based on user identity extracted from custom headers.
+	// If multiple headers are specified, all headers must match for the rule to match.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=256
+	// +notImplementedHide
+	Headers []AuthorizationHeaderMatch `json:"headers,omitempty"`
+}
+
+// AuthorizationHeaderMatch specifies how to match against the value of an HTTP header within a authorization rule.
+type AuthorizationHeaderMatch struct {
+	// Name of the HTTP header.
+	// The header name is case-insensitive unless PreserveHeaderCase is set to true.
+	// For example, "Foo" and "foo" are considered the same header.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"name"`
+
+	// Values are the values that the header must match.
+	// If multiple values are specified, the rule will match if any of the values match.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=256
+	Values []string `json:"values"`
```

Comment:
Only exact matches are supported for now. It should be good enough for authorization use cases. If use cases for other matching types arise, we can add a MatchingType field here.